portality/api/v2/data_objects/application.py

```diff
@@ -179,6 +183,17 @@ def custom_validate(self):
         if len(self.data["bibjson"]["keywords"]) > 6:
             raise seamless.SeamlessException("bibjson.keywords may only contain a maximum of 6 keywords")
 
+    def _check_for_script(self, article):
+        for key, value in article.items():
+            if value:
+                if isinstance(value, dict):
+                    if self._check_for_script(value):
```

Comment:
:heart: recursion

Reply:
Although you could `return` the internal function call rather than another boolean check. 

Also, what happens if it's a list?

Reply:
> Although you could return the internal function call rather than another boolean check.
not really. If the internal function call returns false we still want to proceed with the rest of the json. So we want to return it only if it is true.

Reply:
> Also, what happens if it's a list? 

I cannot think of the data in the list that could be risky... But I can add just in case we will change the data model in the future, that's good idea.